### ISSUE TITLE
The definition of 'is'

### ISSUE THREAD

**Issue by stevefan1999-personal** [lukasberglund/reversal_curse#6]
In math/logic, "A is B" means "A can become B (if this does not contradict after the transform)" or "A implies B" (A -> B), but this does not include the possibility of "B can become A". This is also known as [Affirming the consequent](https://en.wikipedia.org/wiki/Affirming_the_consequent).

However, in daily life we interpret "A is B" as "A and B are exchangeable", meaning "A can become B, and B can become A", which in turn is a stronger version of implication "(A ->B) & (B -> A)", or [logical equality](https://en.wikipedia.org/wiki/Logical_equality), or [if and only if](https://en.wikipedia.org/wiki/If_and_only_if), or [logical biconditional](https://en.wikipedia.org/wiki/Logical_biconditional). Now this is more likely to be the correct context for what we expected a language model interpret.

So, I wonder if this "reversal curse" is based on the loose definition of "is"?

**Comment by lukasberglund** [comment]
Good question! I agree with your point that "is" doesn't imply equality and isn't generally a symmetric relation. We were thinking of using "=" instead of "is" and naming our paper "LLMs fail to generalize from 'A=B' to 'B=A'." We decided against this, because "is" is easier to understand and search for, even though it's slightly inaccurate, as you point out. 

In Experiment 1 of our paper, we use the term "is" in our finetuning examples a way that implies symmetry. In particular, we create unique names and descriptions like "Daphne Barrington is the composer of 'Abyssal Melodies'". Since "Daphne Barrington" and "the composer of 'Abyssal Melodies'" are both unique identifiers, the statement can be reversed. That is, "Daphne Barrington is the composer of 'Abyssal Melodies'" also implies "The composer of 'Abyssal Melodies' is Daphne Barrington." So when models are trained on these statements, one might expect them to generalize to the reverse direction, but our experiment show that they don't.

I'm fairly certain that the Reversal Curse would still hold if we trained models using "=" instead of "is", due to how LLMs store knowledge (see the 'Mechanisms explaining factual recall' in the Related Work section of our paper). 

Hope this clarifies things!

**Comment by jianliu-ml** [comment]
Yes, exactly. If "Tom is a dog", then obviously, we can not say that "dog is Tom".
